### DISCUSSION TITLE
Is there a way to test if elevated when using gsudo?

### DISCUSSION THREAD

**Discussion by eggbean** [gerardog/gsudo#314] [Q&A] [answered]
[This page](https://www.robvanderwoude.com/clevertricks.php) has this useful bit of CMD code, but it doesn't seem to work when elevated using `gsudo` (it always exits). What could I do instead?
```
::Test If script has Admin Priviledges/is elevated
AT > NUL
IF %ERRORLEVEL% EQU 0 (
    ECHO you are Administrator
) ELSE (
    ECHO you are NOT Administrator. Exiting...
    PING 127.0.0.1 > NUL 2>&1
    EXIT /B 1
)
```

**Comment by eggbean** [comment]
I found a way that works:

```
net session >nul 2>&1
if not %ERRORLEVEL% == 0 (
    echo Not admin/elevated
	exit /b 1
)
```

**Comment by gerardog** [comment, marked as answer]
You got it!

You can also check out these:

with gsudo:
* https://gerardog.github.io/gsudo/docs/tips/script-self-elevation#detect-if-running-elevated
```
gsudo status IsElevated --no-output 
if errorlevel 0 echo Current Process Is Not Elevated
if errorlevel 1 echo Current Process Is Elevated
```
* see more here https://github.com/gerardog/gsudo/tree/master/sample-scripts

without gsudo:
* https://github.com/gerardog/gsudo/blob/master/sample-scripts/self-elevate-without-gsudo.cmd
